evalml/tests/automl_tests/test_automl_iterative_algorithm.py

```diff
@@ -415,9 +415,7 @@ def test_max_batches_num_pipelines(
         # automl_algorithm will include all allowed_pipelines in the first batch even
         # if they are not searched over. That is why n_automl_pipelines does not equal
         # n_results when max_iterations and max_batches are None
-        n_automl_pipelines = (
-            1 + len(automl.allowed_pipelines) + automl._pipelines_per_batch
-        )
+        n_automl_pipelines = 1 + len(automl.allowed_pipelines)
```

Comment:
first batch should only be baseline + a full round of possible estimators. What we had previously was erroneous due to the `should_continue()` and `max_batches` logic I mentioned above.